src/nvim/screen.c

```diff
@@ -5887,6 +5887,18 @@ static void screen_char(unsigned off, int row, int col)
   }
 
   ui_cursor_goto(row, col);
+
+  if (p_wd) {
+    if (!got_int) {
+      ui_set_highlight(ScreenAttrs[off] ^ HL_STANDOUT);
+      ui_putc('X');
+      ui_cursor_goto(row, col);
```

Comment:
ui_cursor_goto seems to hide the effects (if any) of  HL_STANDOUT/`ui_putc('X')`. The cursor placement helps  visually, so I don't think HL_STANDOUT/`ui_putc('X')` is needed.

Also, I tested this locally, and moved it back to `ui_puts()` and it seems to work fine there, so it should stay there.

The key change is the checks for `got_int` . Let's move forward with that.

Reply:
> ui_cursor_goto seems to hide the effects (if any) of HL_STANDOUT/ui_putc('X')

Depends on the cursor shape (IIRC).
But I agree that it is not necessary really, and using a block cursor would be the better improvement (but is not necessary).